src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -196,7 +198,7 @@ public function render($request, Exception $e)
      */
     protected function prepareException(Exception $e)
     {
-        if ($e instanceof ModelNotFoundException) {
+        if ($e instanceof ModelNotFoundException || $e instanceof SuspiciousOperationException) {
```

Comment:
You should not propagate the exception message from `SuspiciousOperationException` to `NotFoundHttpException` since we don't want to show it to users.

Reply:
Unless, we do? Is the message you used in the tests actually realistic?

Reply:
I believe we do. The 404 response was returned until the change was made by Symfony in April. The current 500 seems like an unhandled error. We see these attempts quite often. I believe attackers tries to scan for vulnerabilities in routing functionality (with no success in this case).
I found the same error reported here: https://github.com/octobercms/october/issues/4359

Reply:
Right, in which case we don't want to pass along the real exception message. Just having it available as the previous exception is enough. It is often the case that people will show the exception message of httpnotfoundexception to users, so we don't want to show anything that is not meant for users.

Reply:
Sorry, I got your first reply wrong. I didn't thought out users showing the actual exception message, so you're probably right that we shouldn't pass the message a long.
I'll put this in a separate statement below the others with a `null` message.

Reply:
I updated this PR to prevent propagating the exception message.